addons/mail/static/src/core/messaging_service.js

```diff
@@ -251,10 +250,15 @@ export class Messaging {
                         type: notif.payload.channel.channel_type,
                     });
                     break;
-                case "mail.channel/transient_message":
-                    return this.messageService.createTransient(
+                case "mail.channel/transient_message": {
+                    const channel =
+                        this.store.threads[createLocalId("mail.channel", notif.payload.res_id)];
+                    const message = this.messageService.createTransient(
                         Object.assign(notif.payload, { body: markup(notif.payload.body) })
                     );
+                    channel.messages.push(message);
```

Comment:
Wouldn't this create a gap if the most recent are not loaded?

Reply:
And/or wouldn't the transient be gone when returning to most recent?

Reply:
Indeed, transient messages should be stored in their own list `transientMessages`,  so that they can be reconciled when message list changes

Reply:
Decided with seb that the juice is not worth the squeeze, resolving the conversation.